dashboard/src/components/AppView/SecretItem.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from "react";
+import { Eye, EyeOff } from "react-feather";
+
+import { ISecret } from "../../shared/types";
+import "./SecretContent.css";
+
+interface ISecretItemProps {
+  secret: ISecret;
+}
+
+interface ISecretItemState {
+  display: { [s: string]: boolean };
+}
+
+class SecretItem extends React.Component<ISecretItemProps, ISecretItemState> {
+  public state: ISecretItemState = {
+    display: {},
+  };
+
+  public componentDidMount() {
+    const display = {};
+    Object.keys(this.props.secret.data).forEach(k => (display[k] = false));
+    this.setState(display);
+  }
+
+  public render() {
+    const { secret } = this.props;
+    let data: JSX.Element[] = [];
+    Object.keys(secret.data).forEach(k => {
+      data = data.concat(this.renderSecret(k));
+    });
+    return (
+      <tr className="flex">
+        <td className="col-2">{secret.metadata.name}</td>
+        <td className="col-2">{secret.type}</td>
+        <td className="col-7 padding-small">{data}</td>
+      </tr>
+    );
+  }
+
+  private renderSecret = (name: string) => {
+    const toggle = () => this.toggleDisplay(name);
+    const text = atob(this.props.secret.data[name]);
+    return (
+      <span className="flex">
+        <a onClick={toggle}>{this.state.display[name] ? <EyeOff /> : <Eye />}</a>
```

Comment:
I would put the eye next to the data to be shown, that means on the right side. Also, instead of showing the data size, I'd  show  "hidden"